File: test-dest/32_nullCoalescing.swift
func main(args: [String]) {
  do {
    let x: Int32? = try 42
    let y: Int32? = try nil

    let x1 = try x ?? 0
    let y1 = try y ?? 0
    let x2 = try x ?! { throw Exception("FAIL: Should never happen") }
    let y2 = try y ?! { throw Exception("SUCCESS: Should happen") }
    try print("Fail")
  } catch {
    print("Success")
  }

  var values: Array<Int32?> = [1, 2, nil, nil, 3, 4, nil]
  for value in values {
    let _kotliftOptional1 = value; if _kotliftOptional1 == nil { continue }; let a = _kotliftOptional1!
    print(a)
  }
}

main([])

File: test-src/32_nullCoalescing.kt
fun main(args: Array<String>) {
  try {
    val x: Int? = 42
    val y: Int? = null

    val x1 = x ?: 0
    val y1 = y ?: 0
    val x2 = x ?: throw Exception("FAIL: Should never happen")
    val y2 = y ?: throw Exception("SUCCESS: Should happen")
    println("Fail")
  } catch (e: Exception) {
    println("Success")
  }

  val values: List<Int?> = arrayListOf(1, 2, null, null, 3, 4, null)
  for (value in values) {
    val a = value ?: continue
    println(a)
  }
}
